DependencyInjection/HttplugExtension.php

```diff
@@ -45,10 +57,22 @@ public function load(array $configs, ContainerBuilder $container)
             }
 
             $def = $container->register('httplug.client.'.$name, DummyClient::class);
-            $def->setFactory([new Reference($arguments['factory']), 'createClient'])
-                ->addArgument($arguments['config']);
+
+            if (empty($arguments['plugins'])) {
```

Comment:
i would keep it. there is no way to add plugins later at runtime, so it would only be relevant for people that typehint the concrete class instead of the interface, which would be so wrong that we should not care about helping them :-)